js/test/cede/test.huobi-private-proxy.js

```diff
@@ -0,0 +1,208 @@
+            return {
+                "status": "ok",
+                "data": [
+                    {
+                        "symbol": "BTC",
+                        "contract_code": "BTC-USDT",
+                        "contract_size": 0.001000000000000000,
+                        "price_tick": 0.100000000000000000,
+                        "delivery_date": "",
+                        "delivery_time": "",
+                        "create_date": "20201021",
+                        "contract_status": 1,
+                        "settlement_date": "1682006400000",
+                        "support_margin_mode": "all",
+                        "business_type": "swap",
+                        "pair": "BTC-USDT",
+                        "contract_type": "swap",
+                        "trade_partition": "USDT"
+                    },
+                ]
+            }
+        case 'https://api.hbdm.com/swap-api/v1/swap_contract_info':
+            return {
+                "status": "ok",
+                "data": [
+                    {
+                        "symbol": "BTC",
+                        "contract_code": "BTC-USD",
+                        "contract_size": 100.000000000000000000,
+                        "price_tick": 0.100000000000000000,
+                        "delivery_time": "",
+                        "create_date": "20200325",
+                        "contract_status": 1,
+                        "settlement_date": "1682006400000"
+                    },
+                ]
+            }
+        default:
+            return {}
+    }
+}
+
+const URL_MUST_CONTAINING_PROXY = {
+    'https://fakeProxy.com/https://api.hbdm.com/linear-swap-api/v3/swap_cross_matchresults_exact': true
+}
+
+class ProxyError extends Error {
+    constructor(message) {
+        super(message);
+    }
+}
+
+class HuobiCustom extends huobi {
+    async fetch(url, method = 'GET', headers = undefined, body = undefined) {
+        const newUrl = this.implodeParams(url, this.omit(this.extend({}, this.urls), this.version));
+        const urlWithoutParams = newUrl.split('?')[0];
+        if (newUrl.includes("https://fakeProxy.com/")) {
+            if (URL_MUST_CONTAINING_PROXY[urlWithoutParams] === undefined) {
+                throw new ProxyError("URL shoun't contain https://fakeProxy.com/");
+            }
+        } else {
+            if (URL_MUST_CONTAINING_PROXY[urlWithoutParams] === true) {
+                throw new ProxyError("URL should contain https://fakeProxy.com/");
+            }
+        }
+
+        return mockCall(url);
+    }
+}
+
+(async () => {
+    // Initialize the custom Binance exchange class
+    const exchange = new HuobiCustom({
+        apiKey: '12345678',
+        secret: '12345678',
+        forcedProxy: 'https://fakeProxy.com/',
+    });
+
+    try {
+        await exchange.fetchMyTrades('BTC/USDT:USDT');
+    } catch (e) {
+        // the exchange will throw an error because our custom fetch method doesn't 
```

Comment:
Why didn't you use jest.spyOn to mock the calls that fail ?
It would allow you to store values and compare them with what is expected, and raise errors only if something out of the test scope happens